test/jdk/sun/security/ssl/EngineArgs/DebugReportsOneExtraByte.java

```diff
@@ -130,21 +126,28 @@
      * Main entry point for this test.
      */
     public static void main(String args[]) throws Exception {
-        if (debug) {
-            System.setProperty("javax.net.debug", "all");
-        }
 
-        DebugReportsOneExtraByte test = new DebugReportsOneExtraByte();
-        test.runTest();
+        if (args.length == 0) {
+            OutputAnalyzer output = ProcessTools.executeTestJvm(
+                "-Dtest.src=" + System.getProperty("test.src"),
+                "-Djavax.net.debug=all", "DebugReportsOneExtraByte", "p");
+            output.shouldContain("WRITE: TLS10 application_data, length = 8");
 
-        System.out.println("Test Passed.");
+            System.out.println("Test Passed.");
+        } else {
+            DebugReportsOneExtraByte test = new DebugReportsOneExtraByte();
+            test.runTest();
+        }
     }
 
     /*
      * Create an initialized SSLContext to use for these tests.
      */
     public DebugReportsOneExtraByte() throws Exception {
 
+        // Re-enable TLSv1 since test depends on it
+        SecurityUtils.removeFromDisabledTlsAlgs("TLSv1");
+
```

Comment:
I would like move this block to the main() method for safe, or have this method private.

Reply:
Ok.